### DISCUSSION TITLE
Renovate PRs are created with disabled auto-merged. Even though I have set auto merge to true

### DISCUSSION THREAD

**Discussion by shiramax** [renovatebot/renovate#27624] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

Hi!
I'm trying to auto approve and auto-merge renovate's PR's,
Here is my renovate.json file:
```

{
    "extends": [
      "github><github repo>"
    ],
  "commitMessageTopic": "{{packageUpdateTopic}}",
  "semanticCommits": "auto",
  "schedule": "0 * * * *",

  "packageRules": [
    {
      "separateMajorMinor": true,
      "matchDatasources": ["pypi"],
      "registryUrls": ["https://<url>"],
      "matchUpdateTypes": ["patch", "minor", "major"],
      "commitMessageTopic": "{{packageUpdateTopic}}",
      "semanticCommits": true,
      "automerge": true,
      "ignoreDeps": [
        "pydash",
        "dnspython",
        "eventlet",
        "capybara-py"
      ]
    }
    ],
  "platformAutomerge": true,
  "regexManagers": [
    {
      "fileMatch": [
        "requirements(_.+)?\\.txt"
      ],
      "datasourceTemplate": "github-releases",
      "matchStrings": [
        ".+\\/<my org>\\/(?<depName>.*?)\\.git@(?<currentValue>.*)#egg.+"
      ],
      "depNameTemplate": "<Org>/{{depName}}"
    }
  ],
  "docker-compose": {
    "fileMatch": [
      "(^|/)common-services\\.ya?ml$"
    ]
  },
  "hostRules": [{
    "hostName": "<hostname>",
    "authType": "Basic",
    "username": "",
    "encrypted": {
      "password": "<password>"
    }
  }]
}

```
My branch has protection rules:  Require approvals and Require status checks to pass before merging
So I Allowed auto-merge in the repo and also gave permissions to renovate bot and renovate-approve bot 
 
 The checks  for my PR have passed  and I don't see anything special in the logs that might help
 
 Thanks in advanced !

### Logs (if relevant)

<details><summary>Logs</summary>

```

DEBUG: Creating PR (repository=<org>/<repo>, branch=renovate/ski-1.x)
       "prTitle": "Update dependency ski to v1.0.298"
DEBUG: Creating PR (repository=<org>/<repo> branch=renovate/ski-1.x)
       "title": "Update dependency ski to v1.0.298",
       "head": "<org>:renovate/ski-1.x",
       "base": "master",
       "draft": false
DEBUG: PR created (repository=<org>/<repo> branch=renovate/ski-1.x)
       "pr": 558,
       "draft": false
DEBUG: Adding labels '' to #558 (repository=<Org>/<repo>, branch=renovate/ski-1.x)
 INFO: PR created (repository=<org>/<repo>, branch=renovate/ski-1.x)
       "pr": 558,
       "prTitle": "Update dependency ski to v1.0.298"
DEBUG: addParticipants(pr=558) (repository=<org>/<repo>, branch=renovate/ski-1.x)
DEBUG: setPrCache() (repository=<org>/<repo>, branch=renovate/ski-1.x)
DEBUG: Created Pull Request #558 (repository=<org>/<repo>, branch=renovate/ski-1.x)
DEBUG: PR is not configured for automerge (repository=<org>/<repo>, branch=renovate/ski-1.x)

```

</details>

**Comment by rarkins** [comment]
There are multiple things wrong with your package rule. Please describe in plain english what you're trying to achieve with that rule and then I'll tell you what to change.

**Reply by shiramax** [reply]
Thanks!
I want to bump minor, major and patch versions, and auto-merge all the packages that come from my organization, (only if the checks run). I also need auto approve bot to work since my repo requires approval

**Reply by rarkins** [reply]
`ignoreDeps` is not valid within packageRules. What did you want to achieve with that?
- never update these deps, or
- never automerge them

**Reply by shiramax** [reply]
never update these

**Reply by rarkins** [reply]
What as the purpose of including `registryUrls` inside the package rule?

**Reply by shiramax** [reply]
Oh I see it should be here as well, but it's URL for private  package indexer

**Reply by rarkins** [reply]
I don't understand why you added it, so then I can't tell you if or how to remove it

**Reply by shiramax** [reply]
mmm. ok I removed it and it doesn't seems like it is needed, also I changed the ignoreDeps to be outside the packageRules, 
Should those actions reflects the automerge? because I see the PR's creating, just the automerge is still disabled.

thanks again

**Reply by rarkins** [reply]
Can you copy/paste in your latest config?

**Reply by shiramax** [reply]
```

{
    "extends": [
      "github><github repo>"
    ],
  "commitMessageTopic": "{{packageUpdateTopic}}",
  "semanticCommits": "auto",
  "schedule": "0 * * * *",
    "ignoreDeps": [
      "pydash",
      "dnspython",
      "eventlet",
      "capybara-py"
    ],
  "packageRules": [
    {
      "separateMajorMinor": true,
      "matchDatasources": ["pypi"],
      "matchUpdateTypes": ["patch", "minor", "major"],
      "commitMessageTopic": "{{packageUpdateTopic}}",
      "semanticCommits": true,
      "automerge": true
    }
    ],
  "platformAutomerge": true,
  "regexManagers": [
    {
      "fileMatch": [
        "requirements(_.+)?\\.txt"
      ],
      "datasourceTemplate": "github-releases",
      "matchStrings": [
        ".+\\/<my org>\\/(?<depName>.*?)\\.git@(?<currentValue>.*)#egg.+"
      ],
      "depNameTemplate": "<Org>/{{depName}}"
    }
  ],
  "docker-compose": {
    "fileMatch": [
      "(^|/)common-services\\.ya?ml$"
    ]
  },
  "hostRules": [{
    "hostName": "<hostname>",
    "authType": "Basic",
    "username": "",
    "encrypted": {
      "password": "<password>"
    }
  }]
}

```

**Reply by rarkins** [reply]
Comments:

```
  "packageRules": [
    {
      "separateMajorMinor": true, // no need for this setting as it's the default
      "matchDatasources": ["pypi"],
      "matchUpdateTypes": ["patch", "minor", "major"], // not sure you really need this, but you should reconsider if you want to automerge major updates
      "commitMessageTopic": "{{packageUpdateTopic}}", // why?
      "semanticCommits": true, // you usually should put this at the top level of config and not within packageRules, so that it applies to every update
      "automerge": true
    }
    ],
```

Now please capture the debug logs for _one_ branch which you think should automerge, which is not. Take a look yourself or any mention of "automerge"

**Reply by shiramax** [reply]
OK, I removed the commitMessageTopic,  change the semanticCommits position to be in the top level. 
and still 

```
DEBUG: getBranchPr(renovate/ski-1.x) (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: findPr(renovate/ski-1.x, undefined, open) (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: findPr(renovate/ski-1.x, undefined, closed) (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: Found PR #560 (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: branchExists=true (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: dependencyDashboardCheck=undefined (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: Check for closed PR because recreating closed PRs is disabled. (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: findPr(renovate/ski-1.x, Update dependency ski to v1.0.298, !open) (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: prAlreadyExisted=false (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: Checking if PR has been edited (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: branch.isModified(): using git to calculate (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: branch.isModified() = false (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: Checking schedule(, null) (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: No schedule defined (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: Branch already exists (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: branch.isBehindBase(): using git to calculate (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: branch.isBehindBase(): false (repository=Sygnia/moose-server, baseBranch=master, branch=renovate/ski-1.x)
       "branchName": "renovate/ski-1.x"
DEBUG: Branch is up-to-date (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: isBranchConflicted(master, renovate/ski-1.x) (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: branch.isConflicted(): using git to calculate (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: branch.isConflicted(): false (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: Branch does not need rebasing (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: Using reuseExistingBranch: true (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: Setting current branch to master (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: latest commit (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
       "branchName": "master",
       "latestCommitDate": "2024-02-26T17:26:08+02:00"
DEBUG: manager.getUpdatedPackageFiles() reuseExistingBranch=true (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: Branch dep ski in moosegrid/setup.cfg is already updated (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: No content changed (repository=Sygnia/moose-server, packageFile=moosegrid/setup.cfg, branch=renovate/ski-1.x)
       "depName": "ski"
DEBUG: No package files need updating (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: No updated lock files in branch (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: No files to commit (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: Setting current branch to master (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: latest commit (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
       "branchName": "master",
       "latestCommitDate": "2024-02-26T17:26:08+02:00"
DEBUG: Checking if we can automerge branch (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: mergeStatus=no automerge (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: Ensuring PR (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: There are 0 errors and 0 warnings (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: getBranchPr(renovate/ski-1.x) (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: findPr(renovate/ski-1.x, undefined, open) (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: findPr(renovate/ski-1.x, undefined, closed) (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: Found PR #560 (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: getPrCache() (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
 WARN: Package lookup failures (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
       "warnings": [
         "Failed to look up docker package sygniaconsulting/builder-slim",
         "Failed to look up docker package sygniaconsulting/base-node-slim",
         "Failed to look up docker package sygniaconsulting/base-slim",
         "Failed to look up docker package 572055277304.dkr.ecr.us-east-1.amazonaws.com/charts//common",
         "Failed to look up docker package 572055277304.dkr.ecr.us-east-1.amazonaws.com/charts/common"
       ],
       "files": [
         "jupyter_lab/Dockerfile",
         "moosegrid/Dockerfile",
         "deploy/manifests/jupyter-lab/Chart.yaml",
         "deploy/manifests/jupyterhub/Chart.yaml"
       ]
DEBUG: Creating PR (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
       "prTitle": "Update dependency ski to v1.0.298"
DEBUG: Creating PR (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
       "title": "Update dependency ski to v1.0.298",
       "head": "Sygnia:renovate/ski-1.x",
       "base": "master",
       "draft": false
DEBUG: PR created (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
       "pr": 561,
       "draft": false
DEBUG: Adding labels '' to #561 (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
 INFO: PR created (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
       "pr": 561,
       "prTitle": "Update dependency ski to v1.0.298"
DEBUG: addParticipants(pr=561) (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: setPrCache() (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
DEBUG: Created Pull Request #561 (repository=Sygnia/moose-server, branch=renovate/ski-1.x)
**DEBUG: PR is not configured for automerge (repository=Sygnia/moose-server, branch=renovate/ski-1.x)**
```

**Reply by rarkins** [reply]
Sorry, will need the full log from start to finish.

**Reply by shiramax** [reply]
it's super long. can I somehow send it to you privately?

**Reply by rarkins** [reply]
Yes, you can find my email in my github profile

**Reply by rarkins** [reply]
Your log says:
```

DEBUG: CLI config
       "config": {
         "requireConfig": "ignored",
         "schedule": [],
         "token": "***********"
       }

DEBUG: Config file will be ignored (repository=Sygnia/moose-server)
```

Everything we discussed above was pointless because you are running with `--require-config=ignored` and ignoring your own config.

**Reply by shiramax** [reply]
Oh i see,  I did it run my config file locally without the need to merge the changes to the master each time, is there an easy way to do it?

**Reply by rarkins** [reply]
It won't make any difference if you ignore the config. Sorry, but I'm not able to help you further due to a lack of time. This question was invalid due to config error so I will close it.

**Comment by rarkins** [comment, marked as answer]
User set requireConfig=ignored which means all config is ignored so no config at all will work
